flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java

```diff
@@ -173,6 +174,15 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
                     newOptions.remove(removeOption.key());
                 } else if (change instanceof AddColumn) {
                     AddColumn addColumn = (AddColumn) change;
+                    Set<String> fieldNames =
```

Comment:
Why not directly filter the field name of add column in the stream of `newFields`?

Reply:
`Stream.anyMatch`

Reply:
> Why not directly filter the field name of add column in the stream of `newFields`?

It's an interesting question. Maybe that the data types of these two columns are different, so i think it's better to return the error message to the user